name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/payments/PaymentsViewModel.java

```diff
@@ -280,11 +281,11 @@ private void calculate()
         Interval interval = Interval.of(LocalDate.of(startYear - 1, Month.DECEMBER, 31), now);
         Predicate<Transaction> checkIsInInterval = t -> interval.contains(t.getDateTime());
 
+        Set<TransactionPair<?>> transactions = new HashSet<>();
         Map<InvestmentVehicle, Line> vehicle2line = new HashMap<>();
 
         this.sum = new Line(null, false, this.noOfmonths);
         this.sumRetired = new Line(null, useConsolidateRetired, this.noOfmonths);
-        this.transactions = new ArrayList<>();
```

Comment:
Note, the remainder of this method refers to just `transactions`, which got mapped to the member `this.transactions`; but now the same code refers to the new local variable instead.

Reply:
My IDE also hints at this with "Local variables should not shadow class fields" 😉